index.js

```diff
@@ -533,6 +535,10 @@ function Argv (processArgs, cwd) {
     })
   }
 
+  function guessLocale () {
+    return osLocale.sync().split('_')[0]
```

Comment:
I'm thinking it would be better to have y18n fall back to "language only" code (e.g. `es`) if no locale/json file is found for "language_territory" code (e.g. `es_MX`). Then we wouldn't have to worry about doing the split on `'_'` here, which could be prohibitive of adding territory-specific translations to yargs in the future.

What do you think? Perhaps I can put together a PR for y18n to address this.